File: 2021/14/part1.rb
def calculate(file_name)
  lines = File.readlines(file_name)

  template = ""
  rules = {}
  lines.each do |line|
    case line
    when /^([A-Z]+)$/
      template = $1
    when /([A-Z]{2}) -> ([A-Z])/
      rules[$1] = $2
    end
  end
  
  starting = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  30.times do |index|
    puts "#{index}/30 - #{Process.clock_gettime(Process::CLOCK_MONOTONIC)-starting} seconds"
    a = template.chars.each_cons(2).map do |pair|
      mid = rules[pair.join('')]
      pair[0] + mid
    end
    template = a.join('')+template.chars.last
  end
  ending = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  puts "#{ending - starting} seconds"
  
  tally = template.chars.tally
  pp tally
  max = tally.max_by {|k,v| v}[1]
  min = tally.min_by {|k,v| v}[1]
  puts "#{max} - #{min} = #{max-min}"
end

calculate("example_input.txt")
# calculate("input.txt")

File: 2022/1/part1.py
def calculate(file_name):
  """Method"""
  with open(file_name) as file:
    lines = [line.rstrip() for line in file]

  highest_total = 0
  running_total = 0
  for line in lines:
    if line:
      running_total += int(line)
    else:
      highest_total = max(highest_total, running_total)
      running_total = 0
    # print(f"run: {running_total} high: {highest_total}")

  highest_total = max(highest_total, running_total)

  print(highest_total)

calculate("example_input.txt")
calculate("input.txt")